src/main/groovy/com/twcable/gradle/sling/SlingServersConfiguration.groovy

```diff
@@ -15,101 +15,384 @@
+                    it.value.setTheProperty(propName, v)
+                }
+            }
 
-        if (envJsonFilename) {
-            if (!envName) {
-                throw new GradleException("When using the 'envJson' property, you must also specify the 'environment' property")
+        // apply server-specific (non-"env") properties
+        configuration.
+            findAll { serverNameFromConfigurationKey(it.key) != 'env' }.
+            each { String k, def v ->
+                def name = serverNameFromConfigurationKey(k)
+                def propName = serverPropertyFromPropertyKey(k)
+                def serverConfiguration = serverConfs.get(name) // guaranteed to not be null
+                log.info "Setting the ${propName} property of ${name} to ${v}"
+                serverConfiguration.setTheProperty(propName, v)
             }
-            servers = EnvironmentJsonFileReader.getServersFromJsonFile(this, envJsonFilename, envName)
+    }
+
+
+    @Nonnull
+    protected static String serverNameFromConfigurationKey(String propKey) throws GradleException {
+        return extractPropertyGroup(propKey, SERVER_NAME_PATTERN)
+    }
+
+
+    @Nonnull
+    protected static String serverPropertyFromPropertyKey(String propKey) throws GradleException {
+        return extractPropertyGroup(propKey, SERVER_PROPERTY_PATTERN)
+    }
+
+
+    @Nonnull
+    private static String extractPropertyGroup(String propKey, Pattern pattern) throws GradleException {
+        def matcher = pattern.matcher(propKey)
+        def matches = matcher.matches()
+        if (!matches) throw new GradleException("${propKey} does not match the pattern of \"${SLINGSERVER_PROP_PREFIX}{servername}.{propertyname}\"")
+        return matcher.group(1)
+    }
+
+    /**
+     * If the given server name exists in the existing mapping, return it.
+     * Otherwise create a new localhost instance.
+     *
+     * @see #createServerConfiguration(String)
+     */
+    @Nonnull
+    protected static SlingServerConfiguration serverConfForName(String name, ServersMap existing) {
+        if (existing.containsKey(name)) {
+            return existing.get(name)
         }
         else {
-            servers = defaultServers()
+            return createServerConfiguration(name)
         }
     }
 
+    /**
+     * Creates a new localhost instance with uname:pw as admin:admin, using HTTP and the default port, and return it.
+     * If the name has "auth" in it then it is assumed to be an author and use port 4502, otherwise 4503.
+     */
+    @Nonnull
+    static SlingServerConfiguration createServerConfiguration(String name) {
+        if (name.contains("auth")) {
+            log.info "Creating a new author configuration for \"${name}\""
+
+            return new SlingServerConfiguration(
+                name: name, protocol: 'http', port: 4502, machineName: 'localhost',
+                username: 'admin', password: 'admin')
```

Comment:
Absolutely, that's what much of the primary work was about. See the CqBundlePlugin.adoc for how to override with either environment variables or project properties.